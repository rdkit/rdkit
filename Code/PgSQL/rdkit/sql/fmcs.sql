-- github #3687 : crash on bogus fmcs input
select fmcs(m::text) from pgmol where m@>'c1ncccc1';
select fmcs(m) from pgmol where m@>'c1ncccc1';

-- github #3687 : crash on bogus fmcs input
select fmcs('q');
select fmcs('p');
select fmcs('C1CC');
