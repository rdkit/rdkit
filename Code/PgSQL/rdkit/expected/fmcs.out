-- github #3687 : crash on bogus fmcs input
select fmcs(m::text) from pgmol where m@>'c1ncccc1';
                     fmcs                     
----------------------------------------------
 [#6]1:,-[#7]:,-[#6]:,-[#6]:,-[#6]:,-[#6]:,-1
(1 row)

select fmcs(m) from pgmol where m@>'c1ncccc1';
                     fmcs                     
----------------------------------------------
 [#6]1:,-[#6]:,-[#6]:,-[#6]:,-[#6]:,-[#7]:,-1
(1 row)

-- github #3687 : crash on bogus fmcs input
select fmcs('q');
ERROR:  findMCS: could not create molecule from SMILES 'q'
CONTEXT:  SQL function "fmcs_smiles" statement 1
select fmcs('p');
ERROR:  findMCS: could not create molecule from SMILES 'p'
CONTEXT:  SQL function "fmcs_smiles" statement 1
select fmcs('C1CC');
ERROR:  findMCS: could not create molecule from SMILES 'C1CC'
CONTEXT:  SQL function "fmcs_smiles" statement 1
